src/test/unit/shorten-url.test.js

```diff
@@ -22,56 +13,43 @@ function mockFetchForBitly({
   responseFromRequestPayload,
 }: {|
   endpointUrl: string,
-  responseFromRequestPayload: (any) => Response,
+  responseFromRequestPayload: (any) => any,
 |}) {
-  window.fetch.mockImplementation(async (urlString, options) => {
-    const { method, headers, body } = options;
-
-    if (urlString !== endpointUrl) {
-      return new Response(null, {
-        status: 404,
-        statusText: 'Not found',
-      });
-    }
-
-    if (method !== 'POST') {
-      return new Response(null, {
-        status: 405,
-        statusText: 'Method not allowed',
-      });
-    }
-
-    if (
-      headers['Content-Type'] !== 'application/json' ||
-      headers.Accept !== 'application/vnd.firefox-profiler+json;version=1.0'
-    ) {
-      return new Response(null, {
-        status: 406,
-        statusText: 'Not acceptable',
-      });
-    }
-
-    const payload = JSON.parse(body);
-    return responseFromRequestPayload(payload);
-  });
+  window.fetch
+    .catch(404) // catch all
+    .mock(endpointUrl, async (urlString, options) => {
+      const { method, headers, body } = options;
+
+      if (method !== 'POST') {
+        return new Response(null, {
+          status: 405,
+          statusText: 'Method not allowed',
+        });
+      }
+
+      if (
+        headers['Content-Type'] !== 'application/json' ||
+        headers.Accept !== 'application/vnd.firefox-profiler+json;version=1.0'
+      ) {
+        return new Response(null, {
+          status: 406,
+          statusText: 'Not acceptable',
+        });
+      }
+
+      const payload = JSON.parse(body);
+      return responseFromRequestPayload(payload);
```

Comment:
We don't use the functionality to have a response depending on the request body, but I decided to keep it still.